plenum/common/config_util.py

```diff
@@ -17,17 +18,48 @@ def getInstalledConfig(installDir, configFile):
     :return: the configuration as a python object
     """
     configPath = os.path.join(installDir, configFile)
-    if os.path.exists(configPath):
-        spec = spec_from_file_location(configFile, configPath)
-        config = module_from_spec(spec)
-        spec.loader.exec_module(config)
-        return config
-    else:
+    if not os.path.exists(configPath):
         raise FileNotFoundError("No file found at location {}".
                                 format(configPath))
+    spec = spec_from_file_location(configFile, configPath)
+    config = module_from_spec(spec)
+    spec.loader.exec_module(config)
+    return config
 
 
-def getConfig(homeDir=None):
+def extend_with_external_config(extendee: object, extender: Tuple[str, str], required: bool = False) -> object:
+    try:
+        config = getInstalledConfig(*extender)
+        extendee.__dict__.update(config.__dict__)
+    except FileNotFoundError as err:
+        if required:
+            raise err
+    return extendee
+
+
+def extend_with_default_external_config(extendee: object, user_config_dir: str = None) -> object:
+    extend_with_external_config(extendee, (extendee.GENERAL_CONFIG_DIR, extendee.GENERAL_CONFIG_FILE))
```

Comment:
Does it mean that if we have node and CLI on the same machine, then CLI will get the config from /etc?
I think CLI should only get config from ~/.sovrin-cli

Reply:
Will be done in scope of config generalization